### PR TITLE
Throw if action selector returns anything except HTMLElement or Document (close #699)

### LINKED ISSUE #699: We should throw if action selector returns anything except HTMLElement or Document

### PR DESCRIPTION
**Pull request by helen-dikareva**
/cc @inikulin @AndreyBelym @VasilyStrelyaev

### CODE REVIEW

**Comment by AndreyBelym** [comment]
lgtm

File: src/client/core/utils/dom.js [hunk 1]
```diff
@@ -400,6 +400,17 @@ export function findIframeByWindow (iframeWindow, iframeDestinationWindow) {
     return null;
 }
 
+export function getNodeTypeDescription (node) {
+    var nodeType = node.nodeType;
+
+    for (var property in Node) {
+        if (Node.hasOwnProperty(property)) {
+            if (Node[property] === nodeType)
+                return property.toLowerCase().replace('_', ' ');
```
Comment (inikulin) [inline src/client/core/utils/dom.js]: Ehhh, what's going on here?

File: src/client/driver/command-executors/client-functions/selector-executor/index.js [hunk 2]
```diff
@@ -35,6 +35,9 @@ function exists (el) {
 }
 
 function visible (el) {
+    if (!domUtils.isDomElement(el) && !domUtils.isTextNode(el))
```
Comment (inikulin) [inline src/client/driver/command-executors/client-functions/selector-executor/index.js]: Will `isDomElement` return `true` for text node?

Comment (inikulin) [inline src/client/driver/command-executors/client-functions/selector-executor/index.js]: It seems like it should be `||`

**Comment by inikulin** [comment]
\r-

File: src/client/core/utils/dom.js (hunk 1 above)
Comment (helen-dikareva) [inline src/client/core/utils/dom.js]: It's based on `Node type constants` https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType#Node_type_constants

File: src/client/driver/command-executors/client-functions/selector-executor/index.js (hunk 2 above)
Comment (helen-dikareva) [inline src/client/driver/command-executors/client-functions/selector-executor/index.js]: For text node `isDomElement` returns `false`

File: src/client/core/utils/dom.js (hunk 1 above)
Comment (inikulin) [inline src/client/core/utils/dom.js]: Too complicated, can we just have switch by nodeType?

Comment (inikulin) [inline src/client/core/utils/dom.js]: Even `String(node)` should be just fine.

Comment (helen-dikareva) [inline src/client/core/utils/dom.js]: ok

**Comment by VasilyStrelyaev** [comment]
lgtm

File: src/client/driver/command-executors/client-functions/selector-executor/index.js (hunk 2 above)
Comment (inikulin) [inline src/client/driver/command-executors/client-functions/selector-executor/index.js]: What about `document` it should be always visible

**Comment by inikulin** [comment]
lgtm

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by AndreyBelym** [comment]
just the unstable dragging test

**Comment by helen-dikareva** [comment]
please merge

**Pull request merged**